### ISSUE TITLE
Investigate replacing `Newtonsoft.Json` with `System.Text.Json`

### ISSUE THREAD

**Issue by basdijkstra** [basdijkstra/rest-assured-net#99]
This might be a good idea as `System.Text.Json` is built into .NET, whereas `Newtonsoft.Json` is a third party library.

Even better would be if RestAssured.Net can support both, but that would probably be more work.

See https://github.com/eclipse-tractusx/portal-backend/pull/110#issuecomment-1635352855 for the discussion that drives this issue / request.

**Comment by basdijkstra** [comment]
A big blocker for migrating to `System.Text.Json` is the lack of support for JSONPath at the moment: https://github.com/dotnet/runtime/issues/31068.

This would make the kinds of verifications and extractions we do in `Body()` and `Extract()` very cumbersome, if not unusable.

**Comment by basdijkstra** [comment]
Given this comment: https://github.com/dotnet/runtime/issues/31068#issuecomment-1743438493

I'm going to close this one for now and stick with Newtonsoft.Json for the time being.

**Comment by gregsdennis** [comment]
> A big blocker for migrating to System.Text.Json is the lack of support for JSONPath at the moment: https://github.com/dotnet/runtime/issues/31068.

I had posted in that issue that I have a fully-compliant (to the upcoming spec) implementation of JSON Path in [JsonPath.Net](https://www.nuget.org/packages/JsonPath.Net/).  Did that not work for you?

**Comment by basdijkstra** [comment]
@gregsdennis thank you for the comment. I don't know if it works or didn't work, because I haven't tried [JsonPath.Net](https://www.nuget.org/packages/JsonPath.Net/). It looks promising.

What's keeping me from making the migration at the moment is that I'd still be using a third-party library, although less of it as the (de-)serialization part can be handled by `System.Text.Json` itself. Are there any compelling reasons for making the switch, from your perspective?

Also, is there a specific reason there's no `1.0.0` version of JsonPath.Net (yet)? Just curious.

Happy to reopen this for a while.

**Comment by gregsdennis** [comment]
> Are there any compelling reasons for making the switch, from your perspective?

Well, if you need JSON Path functionality with STJ, then it's perfect.  It fully implements the pending spec.

> is there a specific reason there's no 1.0.0 version of JsonPath.Net (yet)?

Yeah, the spec has been pretty much finalized, but it's not ratified into RFC status yet. Once that happens, I'll bump the library to 1.0.0.

**Comment by basdijkstra** [comment]
Got it, thank you. I'll have a look at it soon (probably early January) and decide whether it's worth the effort. I'd be happy to get rid of `Newtonsoft.Json` in favour of STJ. Not that I don't like it, but why use a third-party library when there's a built-in one doing the job?

I could even see this as a first step to full removal once STJ supports JsonPath (which at some point they'll hopefully will).

Thank you once again.

**Comment by gregsdennis** [comment]
To be honest, I'm not sure it will be included given there's a lib that supports it. Maybe.  There's a lot of call for JSON Schema as well (and I have that covered 😁), but the indication from the .net team so far is external support tends to be sufficient until it isn't.

**Comment by basdijkstra** [comment]
Which library do you mean? Yours? Sorry about the confusion here (hey, it's a Saturday morning ;)

**Comment by gregsdennis** [comment]
Yes, I was talking about mine, but it's kinda their approach to any new functionality right now. MS needs some hefty reasoning to warrant the resources to build something like JSON Path or Schema.  Just how it is right now.

**Comment by basdijkstra** [comment]
Gotcha. Thanks. As I said, I'll have a good look at your library and what it would take to implement it.

And thank you, too, for mentioning the JSON Schema capabilities. Not too long ago I moved from `NewtonSoft.Json.Schema` to `NJsonSchema` (because of the licensing), I'm not necessarily looking to replace that yet again, but it might be good to have fewer different sources for third-party features.

Which one of your packages would I need there? All I need is the capability to pass in a schema definition as a `string` or as a typed object (like here: https://github.com/basdijkstra/rest-assured-net/blob/main/RestAssured.Net.Tests/JsonSchemaValidationTests.cs) and check that the actual response in a `HttpResponseMessage` matches the schema. 

See also https://github.com/basdijkstra/rest-assured-net/issues/96

**Comment by gregsdennis** [comment]
For JSON Schema, you can use the `JsonSchema.FromText()` method to get a schema instance... or you can define the schema type-safe-like using the builder.  Here's an [example](https://github.com/gregsdennis/json-everything/blob/master/JsonSchema.Tests/SlackTests.cs#L46-L56).

I have full docs for all my libs at https://docs.json-everything.net.  The main site (https://json-everything.net) is a playground where you can run everything in your browser (Blazor WASM).

**Comment by basdijkstra** [comment]
Got it, thank you very much. As I said, I'll have a look soon, in the next couple of weeks.

**Comment by gregsdennis** [comment]
Json Path is now an RFC!

https://www.rfc-editor.org/rfc/rfc9535.html
